bundles/ch.elexis.core.services/src/ch/elexis/core/services/OrderService.java

```diff
@@ -49,15 +67,145 @@ public List<IOrderEntry> findOrderEntryForStock(IStock stock) {
+				query.or(ModelPackage.Literals.IENCOUNTER__MANDATOR, COMPARATOR.EQUALS, m);
+			}
+			query.andJoinGroups();
+		}
+		for (IEncounter encounter : query.execute()) {
+			for (IBilled billed : encounter.getBilled()) {
+				IBillable billable = billed.getBillable();
+				if (billable instanceof IArticle art) {
+					int amount = (int) billed.getAmount();
+					result.merge(art, amount, Integer::sum);
+				}
+			}
+		}
+		return result;
+	}
+
+	@Override
+	public void reduceOpenEntries(List<IOrder> orders, IArticle article, int reduceBy) {
+		for (IOrder order : orders) {
+			List<IOrderEntry> openEntries = order.getEntries().stream().filter(e -> e.getArticle().equals(article))
+					.filter(e -> e.getState() == OrderEntryState.OPEN).collect(Collectors.toList());
+			for (IOrderEntry entry : openEntries) {
+				if (reduceBy <= 0)
+					return;
+				int amount = entry.getAmount();
+				if (amount <= reduceBy) {
+					reduceBy -= amount;
+					getHistoryService().logRemove(order, entry);
+					modelService.delete(entry);
+				} else {
+					entry.setAmount(amount - reduceBy);
+					modelService.save(entry);
+					getHistoryService().logChangedAmount(order, entry, amount, entry.getAmount());
+					reduceBy = 0;
+				}
+			}
+		}
+	}
+
+	@Override
+	public void createOrderEntries(List<IOrder> existingOrders, IOrder fallbackOrder,
+			Map<IArticle, Integer> entriesToAdd, @Nullable IMandator mandator) {
+		getHistoryService().logCreateOrder(fallbackOrder);
+		for (Map.Entry<IArticle, Integer> entry : entriesToAdd.entrySet()) {
+			if (entry.getValue() <= 0)
+				continue;
+			IArticle article = entry.getKey();
+			int amountToAdd = entry.getValue();
+			boolean appended = false;
+			for (IOrder order : existingOrders) {
+				for (IOrderEntry oe : order.getEntries()) {
+					if (oe.getArticle().equals(article) && oe.getState() == OrderEntryState.OPEN) {
+						int old = oe.getAmount();
+						oe.setAmount(oe.getAmount() + amountToAdd);
+						modelService.save(oe);
+						getHistoryService().logChangedAmount(order, oe, old, oe.getAmount());
+						appended = true;
+						break;
+					}
+				}
+				if (appended)
+					break;
+			}
+			if (!appended) {
+				IStockEntry se = null;
+				if (mandator != null) {
+					String articleStr = StoreToStringServiceHolder.getStoreToString(article);
+					se = StockServiceHolder.get().findPreferredStockEntryForArticle(articleStr, mandator.getId());
+				}
+				IOrderEntry newEntry = (se != null)
+						? fallbackOrder.addEntry(se.getArticle(), se.getStock(), se.getProvider(), amountToAdd)
+						: fallbackOrder.addEntry(article, null, null, amountToAdd);
+				modelService.save(newEntry);
+				getHistoryService().logChangedAmount(fallbackOrder, newEntry, 0, amountToAdd);
+			}
+		}
+		if (!fallbackOrder.getEntries().isEmpty()) {
+			modelService.save(fallbackOrder);
+			modelService.refresh(fallbackOrder, true);
+		}
+	}
+
+	@Override
+	public List<IOrder> findOrderByDate(LocalDate date) {
```

Comment:
Bitte Methode in findOpenOrdersByDate umbenennen.

Der filter auf State == OPEN oder ORDERED wäre performanter wenn das im query wäre und die db erledigt.